File: TradePilot/frontend/src/types/stock.ts
export type MarketCode = 'KOSPI' | 'KOSDAQ' | 'KONEX';

export interface Stock {
  code: string; // "005930" (선행 0 유지)
  name: string;
  market: MarketCode;
  sector?: string;
  market_cap?: number;
}

export interface Quote {
  code: string;
  price: number;
  change: number; // 원 단위
  change_pct: number; // %
  volume: number;
  high?: number;
  low?: number;
  open?: number;
  prev_close?: number;
  ts: number; // epoch ms
  delayed?: boolean;
}

export interface Candle {
  ts: number; // epoch ms
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}

export type CandleInterval = '1m' | '5m' | '15m' | '30m' | 'D' | 'W' | 'M';

export interface OrderbookLevel {
  price: number;
  qty: number;
}

export interface Orderbook {
  code: string;
  ts: number;
  bids: OrderbookLevel[]; // 매수 (높은 가격 순)
  asks: OrderbookLevel[]; // 매도 (낮은 가격 순)
}

File: TradePilot/backend/app/schemas/stock.py
"""종목/시세 도메인 스키마."""
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class StockOut(BaseModel):
    """종목 메타."""

    code: str
    name: str
    market: str
    status: str = "LISTED"
    listing_shares: int | None = None
    market_cap: int | None = None
    listed_at: date | None = None


class StockSearchItem(BaseModel):
    """종목 검색 결과 아이템."""

    code: str
    name: str
    market: str
    sector: str | None = None


class QuoteOut(BaseModel):
    """실시간 시세."""

    code: str
    price: Decimal
    change: Decimal
    change_pct: Decimal
    volume: int
    ts: datetime


class OrderbookLevel(BaseModel):
    """호가 1단계."""

    price: Decimal
    qty: int


class OrderbookOut(BaseModel):
    """호가 10단계."""

    code: str
    asks: list[OrderbookLevel] = Field(default_factory=list)
    bids: list[OrderbookLevel] = Field(default_factory=list)
    ts: datetime


class CandleItem(BaseModel):
    """OHLCV 봉 1행."""

    ts: datetime
    open: Decimal
    high: Decimal
    low: Decimal
    close: Decimal
    volume: int


class FavoriteIn(BaseModel):
    """즐겨찾기 추가 요청."""

    model_config = ConfigDict(extra="forbid")

    code: str = Field(min_length=6, max_length=6)


class FavoriteItem(BaseModel):
    code: str
    name: str
    market: str
    created_at: datetime


CandleInterval = Literal["D", "W", "M", "1m", "5m", "15m", "30m", "60m"]
